---
title: "TE_MA_Paradoxus Practice Analyses"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r packages}
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("Sushi")
# library(sushi)
# 
# plotManhattan(bedfile=Sushi_GWAS.bed,pvalues=Sushi_GWAS.bed[,5],
#                  col=SushiColors(6),genome=Sushi_hg18_genome,cex=0.75)
# labelgenome(genome=Sushi_hg18_genome,n=4,scale="Mb",
#                  edgeblankfraction=0.20,cex.axis=.5)
# axis(side=2,las=2,tcl=.2)
# mtext("log10(P)",side=2,line=1.75,cex=1,font=2)
# mtext("chromosome",side=1,line=1.75,cex=1,font=2)
# 
# plotBedgraph(Sushi_DNaseI.bedgraph,chrom,chromstart,chromend,colorbycol= SushiColors(5))

library(vcfR)
library(adegenet)
library(adegraphics)
library(pegas)
library(StAMPP)
library(lattice)
library(gplots)
library(ape)
library(ggmap)

vcf <- read.vcfR("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/Practice_Analyses/Sample1.vcf")
head(vcf)
vcf@fix[1:10,1:5]

chrom <- create.chromR(name="sample1_data",vcf=vcf)
plot(chrom)

dp <- extract.gt(vcf,element="DP",as.numeric = TRUE)

pdf("DP_sample1_data.pdf",width=10,height=3) #boxplot
boxplot(dp, las=3, col=c("#C0C0C0", "#808080"), ylab="Read Depth (DP)",
las=2, cex=0.4, cex.axis=0.5)
dev.off()

pdf("DP_sample1_data_zoom.pdf", width = 10, height=3) # boxplot
boxplot(dp, las=3, col=c("#C0C0C0", "#808080"), ylab="Read Depth (DP)",
las=2, cex=0.4, cex.axis=0.5, ylim=c(0,50))
abline(h=8, col="red")
dev.off()

getwd()

### Basic Manipulation of a SNP matrix (genlight)
# Convert data into adegenet-native format called genlight:
aa.genlight <- vcfR2genlight(vcf,n.cores=1)
locNames(aa.genlight) <- paste(vcf@fix[,1]) # add real SNP names
pop(aa.genlight) <- substr(indNames(aa.genlight),1,3) # add pop names: here "population" (group) names are 1st 3 chars of ind name 

glPlot(aa.genlight)
x <- summary(t(as.matrix(aa.genlight)))
View(x)


```


```{r mutation rate, echo=FALSE}

# H0vD0.total <- data.frame(matrix(ncol = 2, nrow = 2))
# x <- c("H0","D0")
# rownames(H0vD0.total) <- x
# y <- c("SNPs","Indels")
# colnames(H0vD0.total) <- y
# 
# 
# 
# H0vD0.total[1,1] <- 2.23674E-10
# H0vD0.total[1,2] <- 4.91247E-11
# H0vD0.total[2,1] <- 3.44006E-10
# H0vD0.total[2,2] <- 2.67986E-11 
# 
# test <- fisher.test(H0vD0.total, alternative="two.sided", conf.int=TRUE)



```
## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
