---
title: "TE_MA_Paradoxus Practice Analyses"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r packages}
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("Sushi")
# library(sushi)
# 
# plotManhattan(bedfile=Sushi_GWAS.bed,pvalues=Sushi_GWAS.bed[,5],
#                  col=SushiColors(6),genome=Sushi_hg18_genome,cex=0.75)
# labelgenome(genome=Sushi_hg18_genome,n=4,scale="Mb",
#                  edgeblankfraction=0.20,cex.axis=.5)
# axis(side=2,las=2,tcl=.2)
# mtext("log10(P)",side=2,line=1.75,cex=1,font=2)
# mtext("chromosome",side=1,line=1.75,cex=1,font=2)
# 
# plotBedgraph(Sushi_DNaseI.bedgraph,chrom,chromstart,chromend,colorbycol= SushiColors(5))

library(vcfR)
library(adegenet)
library(adegraphics)
library(pegas)
library(StAMPP)
library(lattice)
library(gplots)
library(ape)
library(ggmap)
library(pinfsc50)
library(reshape2)
library(ggplot2)

#### Their data 
# Find and input VCF data.
# vcf <- system.file("extdata", "pinf_sc50.vcf.gz", package = "pinfsc50")
# vcf <- vcfR::read.vcfR(vcf, verbose = FALSE)
# 
# # Parse DP from the gt region.
# dp <- extract.gt(vcf, element="DP", as.numeric = TRUE)
# 
# # Reorganize and render violin plots.
# dpf <- melt(dp, varnames=c("Index", "Sample"), value.name = "Depth", na.rm=TRUE)
# dpf <- dpf[ dpf$Depth > 0,]
# p <- ggplot(dpf, aes(x=Sample, y=Depth)) + geom_violin(fill="#C0C0C0", adjust=1.0,
#                                                      scale = "count", trim=TRUE)
# 
# p <-  p + theme_bw()
# p <- p + ylab("Read Depth (DP)")
# p <- p + theme(axis.title.x = element_blank(),
#              axis.text.x = element_text(angle = 60, hjust = 1))
# p <- p + stat_summary(fun.data=mean_sdl, geom="pointrange", color="black")
# p <- p + scale_y_continuous(trans=scales::log2_trans(), breaks=c(1, 10, 100, 1000))
# p 
# 
# # Plot as heatmap 
# heatmap.bp(dp[501:1500,])
# 
# # visualization of variant quality 
# 
# vcf_file <- system.file("extdata", "pinf_sc50.vcf.gz",
# package = "pinfsc50")
# dna_file <- system.file("extdata","pinf_sc50.fasta",package="pinfsc50")
# gff_file <- system.file("extdata","pinf_sc50.gff",package="pinfsc50")
# 
# # Read data into memory 
# vcf <- read.vcfR(vcf_file)
# dna <- ape::read.dna(dna_file, format = "fasta")
# gff <- read.table(gff_file, sep="\t", quote="")
# 
# # Create a chromR plot
# chrom <- create.chromR(name="Supercontig",
#                       vcf=vcf, seq=dna, 
#                       ann=gff)
# # Mask for read depth and mapping quality
# chrom <- masker(chrom, min_QUAL=0, min_DP=350,
#                 max_DP=650, min_MQ=59.5,
#                 max_MQ=60.5)
# chrom <- proc.chromR(chrom)
# 
# # Plot.
# chromoqc(chrom, dp.alpha=20)


########################################
# MY DATA
########################################

vcfD0 <- read.vcfR("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/VCFs/D0/D0_reducedGEM.vcf")
head(vcfD0)

# Parse DP from the gt region.
dpD0 <- extract.gt(vcfD0, element="DP", as.numeric = TRUE)

# Reorganize and render violin plots.
dpfD0 <- melt(dpD0, varnames=c("Index", "Sample"), value.name = "Depth", na.rm=TRUE)
dpfD0 <- dpfD0[ dpfD0$Depth > 0,]
pD0 <- ggplot(dpfD0, aes(x=Sample, y=Depth)) + geom_violin(fill="#C0C0C0", adjust=1.0,
                                                     scale = "count", trim=TRUE)

pD0 <-  pD0 + theme_bw()
pD0 <- pD0 + ylab("Read Depth (DP)")
pD0 <- pD0 + theme(axis.title.x = element_blank(),
             axis.text.x = element_text(angle = 60, hjust = 1))
pD0 <- pD0 + stat_summary(fun.data=mean_sdl, geom="pointrange", color="black")
pD0 <- pD0 + scale_y_continuous(trans=scales::log2_trans(), breaks=c(1, 10, 100, 1000))
pdf("violinD0.pdf")
pD0 
dev.off()


# Plot as heatmap 
pdf("heatmapD0.pdf")
heatmap.bp(dpD0[100:500,])
dev.off()

#########
# visualization of variant quality 
########
dna_fileD0 <- file("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/Ref_Genome/genome.337.fasta")
gff_fileD0 <- file("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/Ref_Genome/337.nuclear_genome.EVM.gff3")

# Read data into memory 
dnaD0 <- ape::read.dna(dna_fileD0, format = "fasta")
gffD0 <- read.table(gff_fileD0, sep="\t", quote="")


# Create a chromR plot
chromD0 <- create.chromR(name="Supercontig",
                      vcf=vcfD0, seq=dnaD0, 
                      ann=gffD0)
# Mask for read depth and mapping quality
chromD0 <- masker(chromD0, min_QUAL=0, min_DP=100,
               max_DP=900)
chromD0 <- proc.chromR(chromD0)

plot(chromD0)

# Plot.
chromoqc(chromD0, dp.alpha=20)






#vcf@fix[1:10,1:5]

chrom <- create.chromR(name="H0_data",vcf=vcf)
plot(chrom)

dp <- extract.gt(vcf,element="DP",as.numeric = TRUE)

pdf("DP_H0_data.pdf",width=10,height=3) #boxplot
boxplot(dp, las=3, col=c("#C0C0C0", "#808080"), ylab="Read Depth (DP)",
las=2, cex=0.4, cex.axis=0.5)
dev.off()

pdf("DP_sample1_data_zoom.pdf", width = 10, height=3) # boxplot
boxplot(dp, las=3, col=c("#C0C0C0", "#808080"), ylab="Read Depth (DP)",
las=2, cex=0.4, cex.axis=0.5, ylim=c(0,50))
abline(h=8, col="red")
dev.off()

getwd()

### Basic Manipulation of a SNP matrix (genlight)
# Convert data into adegenet-native format called genlight:
aa.genlight <- vcfR2genlight(vcf,n.cores=1)
locNames(aa.genlight) <- paste(vcf@fix[,1]) # add real SNP names
pop(aa.genlight) <- substr(indNames(aa.genlight),1,3) # add pop names: here "population" (group) names are 1st 3 chars of ind name 

glPlot(aa.genlight)
x <- summary(t(as.matrix(aa.genlight)))
View(x)


```


```{r mutation rate, echo=FALSE}

# H0vD0.total <- data.frame(matrix(ncol = 2, nrow = 2))
# x <- c("H0","D0")
# rownames(H0vD0.total) <- x
# y <- c("SNPs","Indels")
# colnames(H0vD0.total) <- y
# 
# 
# 
# H0vD0.total[1,1] <- 2.23674E-10
# H0vD0.total[1,2] <- 4.91247E-11
# H0vD0.total[2,1] <- 3.44006E-10
# H0vD0.total[2,2] <- 2.67986E-11 
# 
# test <- fisher.test(H0vD0.total, alternative="two.sided", conf.int=TRUE)



```
## Including Plots

```{r plot bedgraphs}
#file.names -> reads in the files you want to run 

bedgraph.names.H0 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/H0", pattern="*.bedgraph",full.names = TRUE)
bedgraph.names.H0

H0_anc <- bedgraph.names.H0[46] 
H0_anc <- read.table(H0_anc)
dim(H0_anc)
colnames(H0_anc) <- c("Chromosome","Start","End","Coverage")

plot(H0_anc$Coverage, col=as.factor(H0_anc$Chromosome), ylab="Coverage",ylim=c(0,30000), xlab="Bins (10kb)")
# read.table(H0_anc, sep= "\t", header=FALSE)
# colnames(H0_anc) <- c("Chromosome","Start","End","Coverage")

i=1

for(i in 1:length(bedgraph.names.H0)){
  pdf(paste("",bedgraph.names.H0[i],".CovGraph.pdf", sep=""))
  file <- read.table(bedgraph.names.H0[i], sep= "\t", header=FALSE)
  colnames(file) <- c("Chromosome","Start","End","Coverage")
plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
dev.off()
}

############### separate samples into blocks for ylim value
bedgraph_100.H0 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/H0/100/", pattern="*.bedgraph",full.names = TRUE)
bedgraph_100.H0

for(i in 1:length(bedgraph_100.H0)){
  pdf(paste("",bedgraph_100.H0[i],".CovGraph.pdf", sep=""))
  file <- read.table(bedgraph_100.H0[i], sep= "\t", header=FALSE)
  colnames(file) <- c("Chromosome","Start","End","Coverage")
plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,10000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
dev.off()
}





















bedgraph.names.D1 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D1", pattern="*.bedgraph",full.names = TRUE)

for(i in 1:length(bedgraph.names.D1)){
  pdf(paste("",bedgraph.names.D1[i],".CovGraph.pdf", sep=""))
  file <- read.table(bedgraph.names.D1[i], sep= "\t", header=FALSE)
  colnames(file) <- c("Chromosome","Start","End","Coverage")
plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
dev.off()
}

pdf(paste("",bedgraph.names.D1[35],".CovGraph.pdf", sep=""))
D1_45 <- read.table(bedgraph.names.D1[35], sep= "\t", header=FALSE)
  colnames(D1_45) <- c("Chromosome","Start","End","Coverage")
plot(D1_45$Coverage, col=as.factor(D1_45$Chromosome), ylab="Coverage",ylim=c(0,1340000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
dev.off()



bedgraph.names.D0 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D0", pattern="*.bedgraph",full.names = TRUE)

for(i in 1:length(bedgraph.names.D0)){
  pdf(paste("",bedgraph.names.D0[i],".CovGraph.pdf", sep=""))
  file <- read.table(bedgraph.names.D0[i], sep= "\t", header=FALSE)
  colnames(file) <- c("Chromosome","Start","End","Coverage")
plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
dev.off()
}

bedgraph.names.D20 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D20", pattern="*.bedgraph",full.names = TRUE)

for(i in 1:length(bedgraph.names.D20)){
  pdf(paste("",bedgraph.names.D20[i],".CovGraph.pdf", sep=""))
  file <- read.table(bedgraph.names.D20[i], sep= "\t", header=FALSE)
  colnames(file) <- c("Chromosome","Start","End","Coverage")
plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
dev.off()
}

#if you want to try loop: 

#for(i in 1:length(file.names)){
  
#  pdf(paste("Graphs",file.names[i],".CovGraph.pdf", sep=""))
#  files <- read.table(file.names[i], sep= "\t", header=TRUE)
#  plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,30000), xlab="Bins (10kb)")
#  dev.off()
#}



```

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
