---
title: "TE_MA_Paradoxus Analyses"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r packages, echo=FALSE}
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("Sushi")
# library(sushi)

library(vcfR)
library(adegenet)
library(adegraphics)
library(pegas)
library(StAMPP)
library(lattice)
library(gplots)
library(ape)
library(ggmap)
library(pinfsc50)
library(reshape2)
library(ggplot2)
library(karyoploteR)
library(GenomicRanges)

```

VCF Analysis using vcfR 
```{r VCF analysis, include=TRUE,echo=FALSE}
#### Their data 
# Find and input VCF data.
# vcf <- system.file("extdata", "pinf_sc50.vcf.gz", package = "pinfsc50")
# vcf <- vcfR::read.vcfR(vcf, verbose = FALSE)
# 
# # Parse DP from the gt region.
# dp <- extract.gt(vcf, element="DP", as.numeric = TRUE)
# 
# # Reorganize and render violin plots.
# dpf <- melt(dp, varnames=c("Index", "Sample"), value.name = "Depth", na.rm=TRUE)
# dpf <- dpf[ dpf$Depth > 0,]
# p <- ggplot(dpf, aes(x=Sample, y=Depth)) + geom_violin(fill="#C0C0C0", adjust=1.0,
#                                                      scale = "count", trim=TRUE)
# 
# p <-  p + theme_bw()
# p <- p + ylab("Read Depth (DP)")
# p <- p + theme(axis.title.x = element_blank(),
#              axis.text.x = element_text(angle = 60, hjust = 1))
# p <- p + stat_summary(fun.data=mean_sdl, geom="pointrange", color="black")
# p <- p + scale_y_continuous(trans=scales::log2_trans(), breaks=c(1, 10, 100, 1000))
# p 
# 
# # Plot as heatmap 
# heatmap.bp(dp[501:1500,])
# 
# # visualization of variant quality 
# 
# vcf_file <- system.file("extdata", "pinf_sc50.vcf.gz",
# package = "pinfsc50")
# dna_file <- system.file("extdata","pinf_sc50.fasta",package="pinfsc50")
# gff_file <- system.file("extdata","pinf_sc50.gff",package="pinfsc50")
# 
# # Read data into memory 
# vcf <- read.vcfR(vcf_file)
# dna <- ape::read.dna(dna_file, format = "fasta")
# gff <- read.table(gff_file, sep="\t", quote="")
# 
# # Create a chromR plot
# chrom <- create.chromR(name="Supercontig",
#                       vcf=vcf, seq=dna, 
#                       ann=gff)
# # Mask for read depth and mapping quality
# chrom <- masker(chrom, min_QUAL=0, min_DP=350,
#                 max_DP=650, min_MQ=59.5,
#                 max_MQ=60.5)
# chrom <- proc.chromR(chrom)
# 
# # Plot.
# chromoqc(chrom, dp.alpha=20)


########################################
# MY DATA
########################################

# vcfD0 <- read.vcfR("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/VCFs/D0/D0_reducedGEM.vcf")
# head(vcfD0)
# 
# # Parse DP from the gt region.
# dpD0 <- extract.gt(vcfD0, element="DP", as.numeric = TRUE)
# 
# # Reorganize and render violin plots.
# dpfD0 <- melt(dpD0, varnames=c("Index", "Sample"), value.name = "Depth", na.rm=TRUE)
# dpfD0 <- dpfD0[ dpfD0$Depth > 0,]
# pD0 <- ggplot(dpfD0, aes(x=Sample, y=Depth)) + geom_violin(fill="#C0C0C0", adjust=1.0,
#                                                      scale = "count", trim=TRUE)
# 
# pD0 <-  pD0 + theme_bw()
# pD0 <- pD0 + ylab("Read Depth (DP)")
# pD0 <- pD0 + theme(axis.title.x = element_blank(),
#              axis.text.x = element_text(angle = 60, hjust = 1))
# pD0 <- pD0 + stat_summary(fun.data=mean_sdl, geom="pointrange", color="black")
# pD0 <- pD0 + scale_y_continuous(trans=scales::log2_trans(), breaks=c(1, 10, 100, 1000))
# pdf("violinD0.pdf")
# pD0 
# dev.off()
# 
# 
# # Plot as heatmap 
# pdf("heatmapD0.pdf")
# heatmap.bp(dpD0[100:500,])
# dev.off()
# 
# #########
# # visualization of variant quality 
# ########
# dna_fileD0 <- file("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/Ref_Genome/genome.337.fasta")
# gff_fileD0 <- file("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/Ref_Genome/337.nuclear_genome.EVM.gff3")
# 
# # Read data into memory 
# dnaD0 <- ape::read.dna(dna_fileD0, format = "fasta")
# gffD0 <- read.table(gff_fileD0, sep="\t", quote="")
# 
# 
# # Create a chromR plot
# chromD0 <- create.chromR(name="Supercontig",
#                       vcf=vcfD0, seq=dnaD0, 
#                       ann=gffD0)
# # Mask for read depth and mapping quality
# chromD0 <- masker(chromD0, min_QUAL=0, min_DP=100,
#                max_DP=900)
# chromD0 <- proc.chromR(chromD0)
# 
# plot(chromD0)
# 
# # Plot.
# chromoqc(chromD0, dp.alpha=20)
# 
# 
# #vcf@fix[1:10,1:5]
# 
# chrom <- create.chromR(name="H0_data",vcf=vcf)
# plot(chrom)
# 
# dp <- extract.gt(vcf,element="DP",as.numeric = TRUE)
# 
# pdf("DP_H0_data.pdf",width=10,height=3) #boxplot
# boxplot(dp, las=3, col=c("#C0C0C0", "#808080"), ylab="Read Depth (DP)",
# las=2, cex=0.4, cex.axis=0.5)
# dev.off()
# 
# pdf("DP_sample1_data_zoom.pdf", width = 10, height=3) # boxplot
# boxplot(dp, las=3, col=c("#C0C0C0", "#808080"), ylab="Read Depth (DP)",
# las=2, cex=0.4, cex.axis=0.5, ylim=c(0,50))
# abline(h=8, col="red")
# dev.off()
# 
# getwd()
# 
# ### Basic Manipulation of a SNP matrix (genlight)
# # Convert data into adegenet-native format called genlight:
# aa.genlight <- vcfR2genlight(vcf,n.cores=1)
# locNames(aa.genlight) <- paste(vcf@fix[,1]) # add real SNP names
# pop(aa.genlight) <- substr(indNames(aa.genlight),1,3) # add pop names: here "population" (group) names are 1st 3 chars of ind name 
# 
# glPlot(aa.genlight)
# x <- summary(t(as.matrix(aa.genlight)))
# View(x)



#### Load in data for H0 
# H0_mut <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/VCFs/H0/H0_vcf_all_NEWCALLSv2_HM.csv",header=TRUE,fill=TRUE,sep=",")



```

Fisher Test for Mutation Rate 
```{r mutation rate, echo=FALSE}

# H0vD0.total <- data.frame(matrix(ncol = 2, nrow = 2))
# x <- c("H0","D0")
# rownames(H0vD0.total) <- x
# y <- c("SNPs","Indels")
# colnames(H0vD0.total) <- y
# 
# 
# 
# H0vD0.total[1,1] <- 2.23674E-10
# H0vD0.total[1,2] <- 4.91247E-11
# H0vD0.total[2,1] <- 3.44006E-10
# H0vD0.total[2,2] <- 2.67986E-11 
# 
# test <- fisher.test(H0vD0.total, alternative="two.sided", conf.int=TRUE)


```

Bedgraph Plots for Coverage - Old Style 
```{r plot bedgraphs,echo=FALSE}
#file.names -> reads in the files you want to run 
# 
# bedgraph.names.H0 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/H0", pattern="*.bedgraph",full.names = TRUE)
# bedgraph.names.H0

# H0_anc <- bedgraph.names.H0[46] 
# H0_anc <- read.table(H0_anc)
# dim(H0_anc)
# colnames(H0_anc) <- c("Chromosome","Start","End","Coverage")
# 
# plot(H0_anc$Coverage, col=as.factor(H0_anc$Chromosome), ylab="Coverage",ylim=c(0,30000), xlab="Bins (10kb)")
# # read.table(H0_anc, sep= "\t", header=FALSE)
# # colnames(H0_anc) <- c("Chromosome","Start","End","Coverage")

# i=1

# for(i in 1:length(bedgraph.names.H0)){
#   pdf(paste("",bedgraph.names.H0[i],".CovGraph.pdf", sep=""))
#   file <- read.table(bedgraph.names.H0[i], sep= "\t", header=FALSE)
#   colnames(file) <- c("Chromosome","Start","End","Coverage")
# plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
# dev.off()
# }

############### separate samples into blocks for ylim value
# bedgraph_100.H0 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/H0/100/", pattern="*.bedgraph",full.names = TRUE)
# bedgraph_100.H0
# 
# for(i in 1:length(bedgraph_100.H0)){
#   pdf(paste("",bedgraph_100.H0[i],".CovGraph.pdf", sep=""))
#   file <- read.table(bedgraph_100.H0[i], sep= "\t", header=FALSE)
#   colnames(file) <- c("Chromosome","Start","End","Coverage")
# plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,10000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
# dev.off()
# }


# bedgraph.names.D1 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D1", pattern="*.bedgraph",full.names = TRUE)
# 
# for(i in 1:length(bedgraph.names.D1)){
#   pdf(paste("",bedgraph.names.D1[i],".CovGraph.pdf", sep=""))
#   file <- read.table(bedgraph.names.D1[i], sep= "\t", header=FALSE)
#   colnames(file) <- c("Chromosome","Start","End","Coverage")
# plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
# dev.off()
# }

# pdf(paste("",bedgraph.names.D1[35],".CovGraph.pdf", sep=""))
# D1_45 <- read.table(bedgraph.names.D1[35], sep= "\t", header=FALSE)
#   colnames(D1_45) <- c("Chromosome","Start","End","Coverage")
# plot(D1_45$Coverage, col=as.factor(D1_45$Chromosome), ylab="Coverage",ylim=c(0,1340000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
# dev.off()
# 
# 

# bedgraph.names.D0 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D0", pattern="*.bedgraph",full.names = TRUE)
# 
# for(i in 1:length(bedgraph.names.D0)){
#   pdf(paste("",bedgraph.names.D0[i],".CovGraph.pdf", sep=""))
#   file <- read.table(bedgraph.names.D0[i], sep= "\t", header=FALSE)
#   colnames(file) <- c("Chromosome","Start","End","Coverage")
# plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
# dev.off()
# }
# 
# bedgraph.names.D20 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D20", pattern="*.bedgraph",full.names = TRUE)

# for(i in 1:length(bedgraph.names.D20)){
#   pdf(paste("",bedgraph.names.D20[i],".CovGraph.pdf", sep=""))
#   file <- read.table(bedgraph.names.D20[i], sep= "\t", header=FALSE)
#   colnames(file) <- c("Chromosome","Start","End","Coverage")
# plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
# dev.off()
# }
# 
# #### Try plotting graphs of TE Locations

# bed.names.D20 <- list.files(path = "/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/TELocate/D20", pattern="*.bed",full.names = TRUE)
# 
# test <- read.table(bed.names.D20[1], sep= "\t", header=FALSE)
# plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
# 
# 
# for(i in 1:length(bed.names.D20)){
#   pdf(paste("",bed.names.D20[i],".CovGraph.pdf", sep=""))
#   file <- read.table(bed.names.D20[i], sep= "\t", header=FALSE)
#   colnames(file) <- c("Chromosome","Start","End","Coverage")
# plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,134000),xlim=c(0,1200), xlab="Bins (10kb)",main="Coverage By Chromosome")
# dev.off()
# }


#if you want to try loop: 

#for(i in 1:length(file.names)){
  
#  pdf(paste("Graphs",file.names[i],".CovGraph.pdf", sep=""))
#  files <- read.table(file.names[i], sep= "\t", header=TRUE)
#  plot(file$Coverage, col=as.factor(file$Chromosome), ylab="Coverage",ylim=c(0,30000), xlab="Bins (10kb)")
#  dev.off()
#}

```


```{r karyoploteR, include=TRUE}
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("karyoploteR")


# browseVignettes("karyoploteR")
#Function from Sam 
# make_karyo_Spar337 <- function(mergefile,title) {
  Chr_sizes_GR <- toGRanges("~/Dropbox/McQueary/Paradoxus_MA/Ref_Genome/337_chrm_sizes_red.bed") ## This is just a three column bed file of the chromosome sizes
  gene_bed_GR <- toGRanges("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/Ref_Genome/337_sites_2.bed") ## This is just a three column bed file of the gene positions (just the gene entries from the gff3 file)
  # supergene_GR <- GRanges(seqnames = "lg16",ranges=IRanges(7311525,18123987)) ## I made this GRanges object to delineate the supergene to add a block for it later

chroms <- c("Spar_I_RaGOO", "Spar_II_RaGOO", "Spar_III_RaGOO", "Spar_IV_RaGOO", "Spar_V_RaGOO", "Spar_VI_RaGOO", "Spar_VII_RaGOO", "Spar_VIII_RaGOO", "Spar_IX_RaGOO", "Spar_X_RaGOO", "Spar_XI_RaGOO", "Spar_XII_RaGOO", "Spar_XIII_RaGOO", "Spar_XIV_RaGOO", "Spar_XV_RaGOO","Spar_XVI_RaGOO")
  # kp <- plotKaryotype(genome=Chr_sizes_GR)  
  # plot.params <- getDefaultPlotParams(plot.type = 1) ## This defines how the plot will generally be structured. See the options in the karyoploter documentation
  # plot.params$data1height=8
  # plot.params$data1outmargin = 4
  # plot.params$data1inmargin = 2
  # plot.params$ideogramheight = 2

# D20_A_cov <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D20/D20-A_.bedgraph")
# str(D20_A_cov)
# colnames(D20_A_cov) <- c("chr","start","end","coverage")
# D20_A_GR <- makeGRangesFromDataFrame(D20_A_cov,seqnames.field = "chr", start.field = "start", end.field = "end", keep.extra.columns = TRUE)
#   # ordered <- D20_A_GR[order(D20_A_GR$coverage, na.last = TRUE),] ## Sorts my DEG data to plot the top ten most DE genes later
#   fc.ymax=ceiling(max(abs(D20_A_GR$coverage))) ## Define the maximum axes of the DEG plots
#   fc.ymin=0 ## Define the minimum axes of the DEG plots 
# TEs <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/TELocate/D20/raw/D20-A-R_R1_val_telocate_raw.TY1.bed")
# colnames(TEs) <- c("chr","start","end","TE")
# str(TEs)  
# TEs.2 <- toGRanges(TEs)
# TEs.3 <- toDataframe(TEs.2)


# kpPlotRegions(kp1,data=TEs.2,col="#AACCFF",later.margin=0.01,border=NA,r0=0,r1=0.5)

## Place the DE dots in the region of the plot that I want
  # kpAxis(kp, ymax=fc.ymax, ymin=fc.ymin,side = 2,cex=0.5,r0 = 0.15, r1 = 0.7) ## Add the axis for the DE dots
  # kpAddMainTitle(kp,main=title) ## Add a title to the plot
  # kpPlotRegions(kp,data = supergene_GR,col="red", r0=0,r1=0.1) ## add a box to show where the supergene is
  # kpPlotMarkers(kp, ordered[1:10], labels = ordered[1:10]$Row.names, text.orientation = "vertical",adjust.label.position = TRUE,r0 = 0.75) ## Add markers for the top 10 most DE genes
  # 

## Load in, filter and format the DE data to be plotted. End result should be a GRanges object with extra columns of differential expression stats (logFC, FDR, etc)
  # temp <- mergefile
  # temp$chr <- as.character(temp$chr)
  # temp.sig<-temp[(temp$FDR<=0.01 & grepl("lg",temp$chr)),]
  # temp.sig <- temp.sig[complete.cases(temp.sig), ]
  # DEG_GR <- makeGRangesFromDataFrame(temp.sig, seqnames.field = "chr",start.field = "start",end.field = "end",keep.extra.columns = TRUE)
  # 
  ## Begin generating the plot itself. If you want to change anything in the plot after first generating it, start again here. 
  # kp <- plotKaryotype(genome = Chr_sizes_GR, cytobands = gene_bed_GR,chromosomes = c("Spar_XVI_RaGOO"),plot.params = plot.params)
  # ordered <- DEG_GR[order(DEG_GR$FDR, na.last = TRUE),] ## Sorts my DEG data to plot the top ten most DE genes later
  # fc.ymax=ceiling(max(abs(DEG_GR$logFC))) ## Define the maximum axes of the DEG plots
  # fc.ymin=0 ## Define the minimum axes of the DEG plots 
  # cex.val <- -log10(DEG_GR$FDR)/7 ## Sets the size for each gene dot based on significance
  # col.over <- "#FFBD07AA" #Yellow
  # col.under <- "#00A6EDAA" #Blue
  # col.insig <- "#7F7F7F"
  # sign.col <- rep(col.insig, length(DEG_GR)) ## This and the next two lines generate a an array of color entries linked to the genes to color the dots
  # sign.col[(DEG_GR$logFC<0)&(DEG_GR$FDR<=0.01)] <- col.under
  # sign.col[(DEG_GR$logFC>0)&(DEG_GR$FDR<=0.01)] <- col.over
  
  # kpPoints(kp, data=DEG_GR, y=abs(DEG_GR$logFC), ymax=fc.ymax, ymin=fc.ymin,cex=cex.val, col=sign.col,r0 = 0.15, r1 = 0.7) ## Place the DE dots in the region of the plot that I want
  # kpAxis(kp, ymax=fc.ymax, ymin=fc.ymin,side = 2,cex=0.5,r0 = 0.15, r1 = 0.7) ## Add the axis for the DE dots
  # kpAddMainTitle(kp,main=title) ## Add a title to the plot
  # kpPlotRegions(kp,data = supergene_GR,col="red", r0=0,r1=0.1) ## add a box to show where the supergene is
  # kpPlotMarkers(kp, ordered[1:10], labels = ordered[1:10]$Row.names, text.orientation = "vertical",adjust.label.position = TRUE,r0 = 0.75) ## Add markers for the top 10 most DE genes
  
# } ## Make karyoplot for linearized genome (!!)


# ##################################################################################################
# ##################################################################################################
# # library(Sushi)
# 
# # data(list=c("Sushi_DNaseI.bedgraph","Sushi_ChIPSeq_CTCF.bedgraph",
#         # "Sushi_ChIPExo_CTCF.bedgraph"), package="Sushi")
# 
# D20 = read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D20/D20-A_.bedgraph")
# D20_TE = read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/TELocate/D20/raw/D20-A-R_R1_val_telocate_raw.TY1.bed")
# 
# dd <- list(D20=D20_A_cov)#, D20_TE=D20_TE)
# 
# png("bedgraphPlot.png", width = 1500, height = 1000)
#   kp <- plotKaryotype(zoom="chr1:1-240865",main = "BedGraphs", cex=1)
#   kpAddBaseNumbers(kp, tick.dist = 10e4, add.units = TRUE, cex=0.5)
#   for(i in seq_len(length(dd))) {
#     names(dd[[i]]) <- c("chr", "start", "end", "value") 
#     gr <- toGRanges(dd[[i]])
#     # at <- autotrack(i, length(dd), margin = 0.1)
#     kpPoints(kp, data=gr, ymax=max(gr$value), col=rainbow(10)[i])
#     kpAddLabels(kp, labels = names(dd)[i], cex=0.5)
#   }
# dev.off()
# 
# str(D20)
# 
#     names(D20) <- c("chr", "start", "end", "value") 
#     gr <- toGRanges(D20)
#     # at <- autotrack(D20, length(D20), margin = 0.1)
#     kp <- plotKaryotype(main = "BedGraphs", cex=3)
#     kpAddBaseNumbers(kp, tick.dist = 10e4, add.units = TRUE, cex=1.8)
#     kpPoints(kp, data=gr, ymax=max(gr$value),col=rainbow(10)D20)
#     kpAddLabels(kp, labels = names(D20), r0=at$r0, r1=at$r1)



############## This script works for me ###############
###################################################
### Call karyotype plot 
## Specify which chromosome in this
# load in chromosome sizes and cytobands files


# {kp1 <- plotKaryotype(plot.type=2, genome=Chr_sizes_GR,cytobands=gene_bed_GR,chromosomes = c("Spar_XII_RaGOO"))
# # this tells the program to add base numbers (self-explanatory)
# kpAddBaseNumbers(kp1)
# # this plots whatever markers you tell it to - in this case, markers of TEs in this genome
# kpPlotMarkers(kp1, chr=TEs$chr, x=TEs$start, labels=TEs$TE,data.panel=2)
# # this plots the coverage in 10kb chunks bc that's what we computed
# kpBars(kp1, data=D20_A_GR, y1=D20_A_GR$coverage, ymax=max(D20_A_GR$coverage)/2, data.panel=1,col="blue")}
# # gives a plot of chromosome XII with coverage on top and TE locations on the bottom 


## check out D20-1, look at chr VI first, then chr XVI


# D20_1_cov <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D20/HM-D20-1.bedgraph")
# str(D20_1_cov)
# colnames(D20_1_cov) <- c("chr","start","end","coverage")
# D20_1_GR <- makeGRangesFromDataFrame(D20_1_cov,seqnames.field = "chr", start.field = "start", end.field = "end", keep.extra.columns = TRUE)
# TEs_1 <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/TELocate/D20/raw/HM-D20-1_R1_001_val_telocate.bed")
# colnames(TEs_1) <- c("chr","start","end","TE")
# str(TEs_1)  
# TEs_1 <- toGRanges(TEs_1)
# TEs.3 <- toDataframe(TEs.2)

# {kp1 <- plotKaryotype(plot.type=2, genome=Chr_sizes_GR,cytobands=gene_bed_GR,chromosomes = c("Spar_XII_RaGOO"))
# # this tells the program to add base numbers (self-explanatory)
# kpAddBaseNumbers(kp1)
# # this plots whatever markers you tell it to - in this case, markers of TEs in this genome
# kpPlotMarkers(kp1, data=TEs_1, chr=TEs_1$chr, x=TEs_1$start, labels=TEs_1$TE, data.panel=2)
# # this plots the coverage in 10kb chunks bc that's what we computed
# kpBars(kp1, data=D20_1_GR, y1=D20_1_GR$coverage, ymax=max(D20_1_GR$coverage)/2, data.panel=1,col="green", r0=0, r1=0.25)
# }
# # gives a plot of chromosome XII with coverage on top and TE locations on the bottom 
# kpBars(kp1, data=D20_A_GR, y1=D20_A_GR$coverage, ymax=max(D20_A_GR$coverage)/2, data.panel = 1, col="blue",r0=0.25, r1=0.5)
# kpPlotCoverage(kp1, data=D20_A_GR, data.panel = 1, col="blue",r0=0.25, r1=0.5)
# You need to make sure that the chromosome you call in plotKaryotype is present in the TE file and that the chromosome names match (i.e. the TE files don't have _RaGOO at the end of their chromosome names)

####################################################################################################################
#### Trying with .depth file generated by Audrey (this file has depth at every site in the genome )

# D20_10_depth <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/D20/HM-D20-10.depth")
# str(D20_10_depth)
# colnames(D20_10_depth) <- c("chr","pos","coverage")
# D20_10_GR <- makeGRangesFromDataFrame(D20_10_depth,seqnames.field = "chr", start.field = "pos", end.field = "pos", keep.extra.columns = TRUE)
# 
# TEs_10 <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/TELocate/D20/raw/HM-D20-10_R1_001_val_telocate_raw.TY1.bed")
# colnames(TEs_10) <- c("chr","start","end","TE")
# str(TEs_10)  
# TEs_10 <- toGRanges(TEs_10)
# TEs.3 <- toDataframe(TEs.2)

# mid.regs <- createRandomRegions(nregions = 6000, length.mean = 5000000, length.sd = 1000000, non.overlapping = FALSE)


plotDepthTELocations <- function(depthFile,teFile,sample,chroms) {
  # load in depth table 
  depth <- read.table(depthFile)
  #add column names
colnames(depth) <- c("chr","pos","coverage")
# turn file into a GRanges object
depthGR <- makeGRangesFromDataFrame(depth,seqnames.field = "chr", start.field = "pos", end.field = "pos", keep.extra.columns = TRUE)
# load in TE file (from TELocate)
TEs <- teFile
# add column names 
colnames(TEs) <- c("chr","start","end","TE")
# make GRanges object from TE file 
TEs <- toGRanges(TEs)
# make pdf plot for all chromosomes: depth + TE locations 

# {pdf(paste("depthTELocate",sample,chroms,".pdf",sep=""))
  #general plotKaryotype command (same for all samples)
  kp <- plotKaryotype(genome=Chr_sizes_GR, cytobands=gene_bed_GR, chromosomes= chroms, cex=0.5, main = paste(sample,chroms,sep="_"))
# this tells the program to add base numbers (self-explanatory)
  # can specify how spaced out you want them using tick.dist 
  # units adds "kb", "mb", etc 
kpAddBaseNumbers(kp,tick.dist = 100000,add.units = TRUE)
# this plots whatever markers you tell it to - in this case, markers of TEs in this genome
# can specify colors of lines and text, distance labels are from each other, etc 
# data panel = 2 specifies that it is the bottom panel 
kpPlotMarkers(kp, data=TEs, chr=TEs$chr, x=TEs$start, labels=TEs$TE, data.panel=2,cex=0.5, text.orientation = "horizontal", line.color = "pink", label.color = "purple", label.dist = 0.01, max.iter = 1000)
# this plots coverage at every base along each chromosome from depth file 
# data panel = 1 specifies it is the top panel 
kpBars(kp, data=depthGR, y1=depthGR$coverage, ymax=max(depthGR$coverage)/2, data.panel=1)
# dev.off() }

}

depthFileNamesD20 <- list.files(path = "/Users/hollymcqueary/Dropbox/depthFiles/D20", pattern="*.depth",full.names = TRUE)

testDepth <- read.table("/Users/hollymcqueary/Dropbox/depthFiles/D20/HM-D20-24.depth")
consensusTEs <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/BedGraphs/TELocate/consensusTEs.txt")
# plotDepthTELocations(testDepth,consensusTEs,"Sample24All","Spar_II_RaGOO")
# plotDepthTELocations(testDepth,consensusTEs,"Sample24All",chroms)

# i=1
sampleNames <- c("Sample1","Sample10","Sample11","Sample12","Sample13","Sample14","Sample15","Sample16","Sample17","Sample18","Sample19","Sample2","Sample20","Sample21","Sample22","Sample23","Sample24","Sample25","Sample26","Sample27","Sample28","Sample29","Sample3","Sample30","Sample31","Sample32","Sample33","Sample34","Sample35","Sample36","Sample37","Sample38","Sample39","Sample4","Sample40","Sample41","Sample42","Sample43","Sample44","Sample45","Sample46","Sample47","Sample48","Sample5","Sample6","Sample8","Sample9","Ancestor")
sampleNames[12]

length(sampleNames)

length(depthFileNamesD20)

plotDepthTELocations(depthFileNamesD20[17],consensusTEs,sampleNames[17],"Spar_II_RaGOO")
plotDepthTELocations(depthFileNamesD20[3],consensusTEs,sampleNames[3],"Spar_X_RaGOO")
plotDepthTELocations(depthFileNamesD20[17],consensusTEs,sampleNames[17],"Spar_IX_RaGOO")
plotDepthTELocations(depthFileNamesD20[3],consensusTEs,sampleNames[3],"Spar_XIV_RaGOO")
plotDepthTELocations(depthFileNamesD20[12],consensusTEs,sampleNames[12],"Spar_XIV_RaGOO")
plotDepthTELocations(depthFileNamesD20[13],consensusTEs,sampleNames[13],"Spar_XV_RaGOO")


for(i in 1:48) {
plotDepthTELocations(depthFileNamesD20[i],consensusTEs,sampleNames[i],"Spar_VI_RaGOO")
}





#, r0=0, r1=0.25)}
# kpBars(kp1, data=D20_A_GR, y1=D20_A_GR$coverage, ymax=max(D20_A_GR$coverage)/2, data.panel = 1, col="blue",r0=0.25, r1=0.5)
# kpPlotCoverage(kp1, data=D20_10_GR, data.panel = 1, col="blue",r0=0.25, r1=0.5)}
# gives a plot of chromosome XII with coverage on top and TE locations on the bottom 

# You need to make sure that the chromosome you call in plotKaryotype is present in the TE file and that the chromosome names match (i.e. the TE files don't have _RaGOO at the end of their chromosome names)
# kpPlotRegions(kp10, data=gene_bed_GR,data.panel = 2)
##### Use kpRect to plot TEs so you can use start and stop positions to show length of TE 
# kpPlotRegions(kp10, data=TEs_10, border=NA, r0=0.2, r1=1, data.panel = 2)



```

Plots 
```{r plots, include=TRUE,echo=FALSE}
h0_triNT <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/VCFs/H0/H0_snps_tricontext.txt",header=TRUE)
colnames(h0_triNT) <- c("alt","context","Relative Mutation Rate","Percent")
h0_0_1 <- subset(h0_triNT, context=="GCG")
h0_0_2 <- subset(h0_triNT, context=="CAG")

library(ggplot2)
h0_triNT_plot <- qplot(x=context,y=`Relative Mutation Rate`, data=h0_triNT, geom="point")
h0_triNT_plot + theme_classic() + theme(axis.text.x = element_text(angle = 90)) + labs(y="Relative Mutation Rate",x="") + scale_y_continuous(n.breaks=13, limits=c(0,14)) + geom_point(data=h0_0_2, colour="red") + geom_point(data=h0_0_1,colour="red")
last_plot()

ggsave("h0_trNT.pdf",width=6,height=4)

h0_triNT_plot <- qplot(x=context,y=Percent, data=h0_triNT, geom="point")
h0_triNT_plot + theme_classic() + theme(axis.text.x = element_text(angle = 90)) + labs(y="Relative Mutation Rate",x="")+ scale_y_continuous(n.breaks=20, limits=c(0,100))
last_plot()


##########################################################
d0_triNT <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/VCFs/D0/D0_snps_context.txt",header=TRUE)
d0_triNT$context <- factor(d0_triNT$context, levels = d0_triNT$context)

library(ggplot2)
d0_triNT_plot <- qplot(x=context,y=Relative_mutation_rate, data=d0_triNT, geom="point")
d0_triNT_plot + theme_classic() + theme(axis.text.x = element_text(angle = 90)) +  aes(x=context,y=Relative_mutation_rate) + labs(y="Relative Mutation Rate",x="") + scale_y_continuous(n.breaks=13, limits=c(0,14))
last_plot()

ggsave("d0_trNT.pdf",width=6,height=4)

h0_triNT_plot <- qplot(x=context,y=Percent, data=h0_triNT, geom="point")
h0_triNT_plot + theme_classic() + theme(axis.text.x = element_text(angle = 90)) + labs(y="Relative Mutation Rate",x="")+ scale_y_continuous(n.breaks=20, limits=c(0,100))
last_plot()


```

Comparison of mutation rates between experiments 
```{r compare, include=TRUE, echo=FALSE}
compare <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Paradoxus_MA/compare_table.txt", header=TRUE)


library(ggplot2)
compare$strain <- c("SparHap0","SparHap0.1","SparHap1","S.pombe","S.cer-Hap","S.cer-Dip","S.cer-DipZhu")
pdf("mutation_rate_comp2.pdf")
compPlot <- ggplot(data=compare, aes(strain,mutation_rate,ymin=mutation_rate-SE,ymax=mutation_rate+SE))
compPlot + geom_pointrange() + theme(axis.text.x = element_text(size = 10)) + ggtitle("Mutation Rate Comparison") + labs(x="Study",y="mutation rate per site per generation")
dev.off()




```




Packages used in this analysis


```{r packagesUsed, include = TRUE }
library(vcfR)
library(adegenet)
library(adegraphics)
library(pegas)
library(StAMPP)
library(lattice)
library(gplots)
library(ape)
library(ggmap)
library(pinfsc50)
library(reshape2)
library(ggplot2)
library(karyoploteR)
library(GenomicRanges)
```


